### PR TITLE
Test cleanup

### PR DESCRIPTION
**Pull request by rowanseymour**
- Reduces repetition in test classes
- Allows template tests to be run as a performance test
- Removes nose as a dependency since we're using py.test (nose isn't maintained any more)
- Fixes Python 3 issue in class based views module
- Adds test coverage generation (we should connect up coveralls.io)
- Consistent names for test classes and modules

Apologies for the awful diff... I guess I changed a lot and renamed files

### CODE REVIEW

**Comment by rowanseymour** [comment]
@Psycojoker I think you can now enable https://coveralls.io/ for this repo, and then we can have any PR fail if the coverage drops

File: hamlpy/test/test_elements.py [hunk 1]
```diff
@@ -1,98 +1,99 @@
 from __future__ import print_function, unicode_literals
 
-from nose.tools import eq_
+import unittest
 
 from hamlpy.elements import Element
 
-class TestElement(object):
```
Comment (rowanseymour) [inline hamlpy/test/test_elements.py]: Entire class has wrong indentation

File: hamlpy/test/templates/filtersPygments16.hamlpy [hunk 2]
```diff
@@ -1,7 +0,0 @@
-:highlight
```
Comment (rowanseymour) [inline hamlpy/test/templates/filtersPygments16.hamlpy]: These files were previously only being tested if pygments 1.6 was installed... which is never if you have latest pygments loaded. I think it's ok to only test against latest pygments.

File: hamlpy/test/test_attribute_dict_parser.py [hunk 3]
```diff
@@ -0,0 +1,108 @@
...
+
+
+class AttributeDictParserTest(unittest.TestCase):
+
+    @staticmethod
+    def _parse(text):
+        return AttributeDictParser(text).parse()
+
+    def test_parse(self):
+        # TODO
```
Comment (rowanseymour) [inline hamlpy/test/test_attribute_dict_parser.py]: Someone appears to have left themselves a TODO note

**Comment by rowanseymour** [comment]
@Psycojoker please take a look when you get a chance

**Comment by rowanseymour** [comment]
Update: figured out I can enable coveralls for this repo, and removed Tox. Tox isn't necessary for multi-python-version testing but I had in there previously because I was figuring we'd need it if we ever test against different versions of Django - but it seems that can all be done straight from travis with a build matrix. Also added a coverage badge to the readme.

**Comment by Psycojoker** [comment]
Things looks fine, sorry for the delay.

**Pull request merged**